engine/src/main/java/org/hibernate/search/engine/backend/work/execution/OperationSubmitter.java

```diff
@@ -16,44 +18,95 @@
  * Interface defining how operation should be submitted to the queue.
  */
 @Incubating
-public enum OperationSubmitter {
+public abstract class OperationSubmitter {
```

Comment:
I pushed a commit to remove the delegation and the `Operation` interface. As far as I can tell, this works just as well. Please have a look at the commit and let's discuss it here if you disagree.

Reply:
Yup, this works too 😃 thanks
A sealed interface/class would've been nice, but well ... let's leave it for some later Search version 😃